.gitignore

```diff
@@ -2,6 +2,10 @@
 /.phpunit.result.cache
 /box.json
 /composer.lock
+/dev-tools/composer.lock
+/dev-tools/bin/
+/dev-tools/phpstan/cache/
+/dev-tools/vendor/
```

Comment:
@julienfalque why do we need it here as well? git should pick nested .gitignore automatically
```
ker@dus:~/github/FriendsOfPHP/PHP-CS-Fixer/dev-tools λ cat .gitignore 
/composer.lock
/bin/
/phpstan/cache/
/vendor/
```